File: library/src/test/java/me/proxer/library/enums/IndustryTypeTest.java
package me.proxer.library.enums;

import me.proxer.library.ProxerTest;
import org.junit.jupiter.api.Test;

import java.io.IOException;

import static org.assertj.core.api.Assertions.assertThat;

class IndustryTypeTest extends ProxerTest {

    @Test
    void testDefault() throws IOException {
        final IndustryType industryType = api.moshi().adapter(IndustryType.class).fromJson("\"publisher\"");

        assertThat(industryType).isSameAs(IndustryType.PUBLISHER);
    }

    @Test
    void testFallback() throws IOException {
        final IndustryType industryType = api.moshi().adapter(IndustryType.class).fromJson("\"xyz\"");

        assertThat(industryType).isSameAs(IndustryType.UNKNOWN);
    }
}

File: library/src/test/kotlin/me/proxer/library/enums/IndustryTypeTest.kt
package me.proxer.library.enums

import com.serjltt.moshi.adapters.FallbackEnum
import com.squareup.moshi.Moshi
import org.amshove.kluent.shouldBe
import org.junit.jupiter.api.Test

/**
 * @author Ruben Gees
 */
class IndustryTypeTest {

    private val adapter = Moshi.Builder()
        .add(FallbackEnum.ADAPTER_FACTORY)
        .build()
        .adapter(IndustryType::class.java)

    @Test
    fun testDefault() {
        adapter.fromJson("\"publisher\"") shouldBe IndustryType.PUBLISHER
    }

    @Test
    fun testFallback() {
        adapter.fromJson("\"xyz\"") shouldBe IndustryType.UNKNOWN
    }
}
